extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/KeycloakPolicyEnforcerConfig.java

```diff
@@ -0,0 +1,192 @@
+package io.quarkus.keycloak.pep;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.keycloak.representations.adapters.config.PolicyEnforcerConfig;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.runtime.annotations.DefaultConverter;
+
+@ConfigRoot(name = "keycloak", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class KeycloakPolicyEnforcerConfig {
+
+    /**
+     * Adapters will make separate HTTP invocations to the Keycloak server to turn an access code into an access token.
+     * This config option defines how many connections to the Keycloak server should be pooled
+     */
+    @ConfigItem(defaultValue = "20")
+    int connectionPoolSize;
+
+    /**
+     * Policy enforcement configuration when using Keycloak Authorization Services
+     */
+    @ConfigItem
+    KeycloakConfigPolicyEnforcer policyEnforcer;
+
+    @ConfigGroup
+    public static class KeycloakConfigPolicyEnforcer {
+
+        /**
+         * Specifies how policies are enforced.
+         */
+        @ConfigItem
+        boolean enable;
+
+        /**
+         * Specifies how policies are enforced.
+         */
+        @ConfigItem(defaultValue = "ENFORCING")
+        String enforcementMode;
+
+        /**
+         * Specifies the paths to protect.
+         */
+        @ConfigItem
+        Map<String, PathConfig> paths;
+
+        /**
+         * Defines how the policy enforcer should track associations between paths in your application and resources defined in
+         * Keycloak.
+         * The cache is needed to avoid unnecessary requests to a Keycloak server by caching associations between paths and
+         * protected resources
+         */
+        @ConfigItem
+        PathCacheConfig pathCache;
+
+        /**
+         * Specifies how the adapter should fetch the server for resources associated with paths in your application. If true,
+         * the
+         * policy
+         * enforcer is going to fetch resources on-demand accordingly with the path being requested
+         */
+        @ConfigItem(defaultValue = "true")
+        boolean lazyLoadPaths;
+
+        /**
+         * Defines a URL where a client request is redirected when an "access denied" message is obtained from the server.
+         * By default, the adapter responds with a 403 HTTP status code
+         */
+        @ConfigItem
+        Optional<String> onDenyRedirectTo;
+
+        /**
+         * Specifies that the adapter uses the UMA protocol.
+         */
+        @ConfigItem
+        boolean userManagedAccess;
```

Comment:
IMHO everything not related to the Keycloak specific policy-enforcer configuration should not be here at all; we've promised `quarkus-oidc` will work with not only KC :-), and UMA is supported not only by Keycloak

Reply:
I see. So we would be providing a separated extension for UMA support if someone demands it ?

Sounds good for me. I can remove UMA support from this extension.

Reply:
@pedroigor hey Pedro, exactly, lets have it done either directly as part of `quarkus-oidc` (it is nor really that fat so may be it should be there) or a new extension like `quarkus-oidc-uma` - so that we can say Quarkus can be UMA protected with any IDP which does it, which will be massive. Let me open an issue to track it, it would probably be a copy and paste for you :-), may be the only minor effort would be required to replace the KC adapter client code with some basic HTTP client....

Reply:
Client-side, UMA is quite straight forward to implement. I agree that a separated extension is the best way.

Reply:
I won't hold up this PR for this, but we should create an issue to track it.

Reply:
@stuartwdouglas we get it in too fast and the users will see the extra properties in the wrong namespace and a new migration headache is on the way